pkg/stream/xprotocol/connpool_multiplex.go

```diff
@@ -158,14 +157,32 @@ func (p *poolMultiplex) NewStream(ctx context.Context, receiver types.StreamRece
 
 	subProtocol := p.connpool.codec.ProtocolName()
 
-	client, _ := p.activeClients[clientIdx].Load(subProtocol)
+	activeClient := func() *activeClientMultiplex {
+		var ac *activeClientMultiplex
+
+		// max loop twice
+		for i := 0; i < 2; i++ {
+			client, _ := p.activeClients[clientIdx].Load(subProtocol)
+			if client == nil {
+				break
+			}
+			ac = client.(*activeClientMultiplex)
+
+			// init a new connection when got goaway from server.
+			if atomic.LoadUint32(&ac.goaway) == 1 {
+				p.init(subProtocol, int(clientIdx))
```

Comment:
we can retry it  when return `types.ConnectionFailure`, no need to wait for tcp to connect.